src/proof/proof_rule.h

```diff
@@ -232,6 +232,40 @@ enum class PfRule : uint32_t
    * above. \endverbatim
    */
   MACRO_SR_PRED_TRANSFORM,
+  /**
+   * \verbatim embed:rst:leading-asterisk
+   * **Builtin theory -- Encode predicate transformation**
+   * .. math::
+   *   \inferrule{F \mid G}{G}
+   * 
+   * where :math:`F` and :math:`G` are equivalent up to their encoding in an
+   * external proof format. This is currently verified by
+   *   :math:`\texttt{RewriteDbNodeConverter::convert}(F) = 
+   * \texttt{RewriteDbNodeConverter::convert}(G)`.
+   * This rule can be treated as a no-op when appropriate in external proof
+   * formats.
+   * \endverbatim
+   */
+  ENCODE_PRED_TRANSFORM,
+  /**
+   * \verbatim embed:rst:leading-asterisk
+   * **Builtin theory -- DSL rewrite**
+   * .. math::
+   *   \inferrule{F_1 \dots F_n \mid id t_1 \dots t_n}{F}
+   * 
+   * where the DSL rewrite rule with the given identifier is
+   * :math:`\forall x_1 \dots x_n. (G_1 \wedge G_n) \Rightarrow G`
+   * where for :math:`i=1, \dots n`, we have that :math:`F_i = \sigma(G_i)`
+   * and :math:`F = \sigma(G)` where :math:`\sigma` is the substitution
+   * :math:`\{x_1\mapsto t_1,\dots,x_n\mapsto t_n\}`.
+   * 
+   * Notice that the application of the substitution takes into account the
+   * possible list semantics of variables :math:`x_1 \ldots x_n`. If
+   * :math:`x_i` is a variable with list semantics, then :math:`t_i` denotes a
```

Comment:
clarify what it means to substitute a variable into a list. IIRC for this to be sound it's necessary that x_i is an argument of an n-ary operator and the components of t_i will be added to the arguments of that operator at that position?